### PR TITLE
Add find tag feature

### LINKED ISSUE #5: Locating persons by tag
As a user, I can filter my contacts based on their tags (for all contacts, and favourited contacts), so that I can easily keep track of who I am working with at a specific point of time.

- [ ] Locate persons by their tags

### PR DESCRIPTION
**Pull request by glennljw**
Fixes #5.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public boolean test(Person person) {
+        int numTags = person.getTags().size();
+        boolean hasAllMatches = true;
+
+        for (int i = 0; i < keywords.size(); i++) {
+            String keyword = keywords.get(i);
+            boolean doesKeywordMatchAnyTags = person.getTags().stream()
+                    .anyMatch(tag -> StringUtil.containsWordIgnoreCase(tag.tagName, keyword));
+            hasAllMatches = hasAllMatches && doesKeywordMatchAnyTags;
+        }
```
Comment (takufunkai) [inline src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java:28]: Might want to consdeir using a for-each loop here `for (String keyword : keywords)`.
I think it increases readability and abstraction (which makes it safer)!

File: src/main/java/seedu/address/logic/commands/FindTagCommand.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Finds and lists all contacts in address book whose tag contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FindTagCommand extends Command {
+
+    public static final String COMMAND_WORD = "findtag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all contacts whose tag contain any of "
```
Comment (takufunkai) [inline src/main/java/seedu/address/logic/commands/FindTagCommand.java:17]: Slight grammer error: "whose tags contain"

**Review by takufunkai — CHANGES_REQUESTED** [review]
Minor suggestion to change the for-loop to for-each. Otherwise great job, LGTM!

File: src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java (hunk 1 above)
Comment (glennljw) [reply src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java]: Yup I agree. Will change it! 👌

File: src/main/java/seedu/address/logic/commands/FindTagCommand.java (hunk 2 above)
Comment (glennljw) [reply src/main/java/seedu/address/logic/commands/FindTagCommand.java]: Thanks for picking this out 😌

**Pull request merged**